### DISCUSSION TITLE
Add doublet detection to results

### DISCUSSION THREAD

**Discussion by jashapiro** [AlexsLemonade/OpenScPCA-analysis#364] [Propose a new analysis]
### Proposed analysis

In the ScPCA workflow, we do not currently do any analysis or flagging of doublet cells. Adding flagging of potential doublets could improve data quality for cell typing and many other analyses.

### Scientific goals

The presence of doublet cells disrupt many types of analyses, including cell typing and analysis of differential expression among groups of cells. TO that end it would be good to flag and ultimately remove those cells before downstream analysis.

We should start by exploring doublet identification methods that may be available and applicable to the ScPCA project, and implement one across the atlas.

### Methods or approach

I expect that we will want to use the [`scDblFinder` package](https://bioconductor.org/packages/3.18/scDblFinder).  This method does not require genetic data or cellhashing, which we do not have in most cases, but it should be more robust than applying a maximum count cutoff, as has sometimes been proposed.

There is a good overview and tutorial for the `scDblFinder` package as part of the OSCA book:  https://bioconductor.org/books/release/OSCA.advanced/doublet-detection.html`

I would expect we will begin with a bit of testing and benchmarking of this method and potentially others. After initial analysis, we can implement a robust workflow for flagging and/or removing likely doublet cells.

### Existing modules

None

### Input data

All! If we do use `scDblFinder`, this will be the SCE objects, but other methods may use other systems.

### Scientific literature

Germain _et al._ (2021) Doublet Identification in Single-Cell Sequencing Data Using scDblFinder. https://doi.org/10.12688/f1000research.73600.1.

Dahlin _et al._ (2018) A single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in Kit mutant mice. https://doi.org/10.1182/blood-2017-12-821413

**Comment by sjspielman** [comment]
Just popping in to say that I'm going to start reading up on this analysis, including linked papers and more that may be useful. I'll post back here with my thoughts based on the literature and where we might begin.

**Reply by sjspielman** [reply]
- This [benchmarking paper](https://doi.org/10.1016/j.cels.2020.11.008) has some insights into performance across different methods, but this paper came out just before the current version of `scDblFinder` came out. [A protocol redux of that paper](https://arxiv.org/abs/2101.08860) that does include `scDblFinder` found that indeed it had better performance, both in computational efficiency and accuracy
- Based on the benchmarking in the `scDblFinder` (linked in opening discussion post), I think we should also try out methods from [`scds`](https://bioconductor.org/packages/devel/bioc/vignettes/scds/inst/doc/scds.html).

**Reply by sjspielman** [reply]
It's also worth noting that previous efforts to benchmark these packages have relied on datasets with experimentally-validated doublets, but virtually all datasets are from healthy human and/or mouse. As such, how these methods perform on cancer, let alone pediatric cancer, datasets has not been explored. If possible, we should try to find a (pediatric) cancer scRNA-seq dataset with validated doublet calls to benchmark against. That said, if such a dataset(s) exist, it's not likely there are many datasets, so such an analysis would be limited and not necessarily generalizable.

**Reply by sjspielman** [reply]
Just came across this pre-print from 2023 (does not appear to have been published?) - https://doi.org/10.1101/2023.08.04.552078

Of note is its accompanying website. Here I'm specifically linking their doublet detection page: https://goyallab.github.io/SingletCodeWebsite/doubletDetectionMethods/

**Comment by sjspielman** [comment]
Jotting down a quick thought that is not yet fully fleshed out, but don't want to forget it! Potentially as part of benchmarking (or perhaps a separate analysis in and of itself?), we can also compare doublets to cell type annotations. I would hypothesize an association between cells with unknown labels (which `CellAssign`, but generally not `SingleR`, might assign) and doublets, since doublets - specifically _heterotypic doublets_ - are more likely to have confounding expression signatures which would also potentially preclude cell type annotation. Of course, the fact that these are often cancer cells is a similar confounding effect, so this should also be considered when interpreting results.

**Comment by sjspielman** [comment]
Noting this additional python-based method I hadn't come across yet as something we could try, but generally speaking I'm not sure we have enough cells in most of the processed data to get quite as robust inferences - https://github.com/kostkalab/vaeda
